source/grid/grid_generator.cc

```diff
@@ -3809,36 +3812,121 @@ namespace GridGenerator
     // now form the union of the set of cells
     std::vector<CellData<dim>> cells;
     cells.reserve(triangulation_1.n_cells() + triangulation_2.n_cells());
-    for (typename Triangulation<dim, spacedim>::cell_iterator cell =
-           triangulation_1.begin();
-         cell != triangulation_1.end();
-         ++cell)
+    for (const auto &cell : triangulation_1.cell_iterators())
       {
         CellData<dim> this_cell;
         for (unsigned int v = 0; v < GeometryInfo<dim>::vertices_per_cell; ++v)
           this_cell.vertices[v] = cell->vertex_index(v);
         this_cell.material_id = cell->material_id();
-        cells.push_back(this_cell);
+        this_cell.manifold_id = cell->manifold_id();
+        cells.push_back(std::move(this_cell));
```

Comment:
use `emplace_back`. I *think* (but am not sure) that you can then also omit the `std::move` call.

Reply:
No, `std::move` is necessary to call the move constructor and it doesn't matter whether `push_back` or `emplace_back` is used. (https://stackoverflow.com/questions/26860749/efficiency-of-c11-push-back-with-stdmove-versus-emplace-back-for-already). Also compare https://godbolt.org/g/ebNwx8.

Reply:
great, thank you for checking!